### DISCUSSION TITLE
Adjust Brightness/Lightness including pure black

### DISCUSSION THREAD

**Discussion by ukanuk** [ImageMagick/ImageMagick#6357] [Help] [answered]
### ImageMagick version

7.1.1-8 Q16-HDRI x64

### Operating system, version and so on

Win10 22H2

### Description

How can I increase brightness and reduce saturation _including_ pure black? Ultimately my intent is a batch script to `mogrify` many images such that I can then add black text annotations to them without concern for contrast. I have tried `modulate` in HSL and HSB/HSV, but neither do anything to pure black. 

```
magick .\sample.png -modulate 200,50,100`
magick .\sample.png -define modulate:colorspace=HSB -modulate 200,50,100
```

I have read through https://www.imagemagick.org/Usage/color_mods but the only other options I see also affect hue and contrast.

Here is a sample image (note there is transparency):
![sample](https://github.com/ImageMagick/ImageMagick/assets/59303202/3521ab01-968d-432d-9ba9-3d74a836dbd4)

With the free www.getpaint.net I can easily achieve my desired result for a single image with the dialog for "Adjustments" → "Hue / Saturation". I set saturation at 50% and lightness at "60". Lightness at -100 creates pure black, and lightness at +100 creates pure white. But I cannot use Paint.NET on entire folders of images at once, which is why I am trying to get similar results from ImageMagick.
![sample_Paint NET](https://github.com/ImageMagick/ImageMagick/assets/59303202/4469cdbb-48dd-4318-b036-b148c3729354)

Here are my attempts with ImageMagick in HSL and HSB/HSV. Pure black in original image is untouched in all results. 
![sample_magick_modulate](https://github.com/ImageMagick/ImageMagick/assets/59303202/71338364-973d-4845-9b5b-36bf04eb24d9)

**Comment by snibgo** [comment, marked as answer]
A common method to lighten or darken an image is by multiplying the colour channels by a number. If the number is between 0 and 1, pixels will become more black. Numbers greater than 1 will make pixels lighter (and cause clipping). This leaves black untouched, and has maximum effect at white. You want the opposite. So first negate, then multiply, then negate again. For example:
```
magick forLighten.png -channel RGB -negate -evaluate Multiply 0.2 -negate +channel x.png
```
![x](https://github.com/ImageMagick/ImageMagick/assets/33812368/ad475746-5bb1-49a8-a928-68eadc447b99)

This will naturally reduce saturation as well as lightening the image. If you want, you can also convert to HCL or similar colorspace, multiply the second channel (Chroma) by a number less than one, and convert back to sRGB.
```
magick forLighten.png -channel RGB -negate -evaluate Multiply 0.2 -negate +channel -colorspace HCL -channel 1 -evaluate Multiply 0.5 +channel -colorspace sRGB x2.png
```
![x2](https://github.com/ImageMagick/ImageMagick/assets/33812368/37d7f8ac-10e6-4ba7-bd85-8e2da944b3b8)

**Reply by ukanuk** [reply]
Thank you, I am able to exactly replicate Paint.NET's effect with this! Note the mathematical values conveniently correspond between ImageMagick and Paint.NET, with (1-0.4) and 160% for brightness, and 0.5 and 50% for saturation, respectively.
```
magick.exe .\sample.png -channel RGB -negate -evaluate Multiply 0.4 -negate  +channel -colorspace HSL -channel 1 -evaluate Multiply 0.5 +channel x.png
```

**Comment by snibgo** [comment]
Good stuff.
